animation_nodes/nodes/generic/c_utils.pyx

```diff
@@ -50,39 +54,89 @@ def mixQuaternionLists(QuaternionList quaternionsA, QuaternionList quaternionsB,
-        results.data[i].y = lerp(quaternionA.y, quaternionB.y, influence)
-        results.data[i].z = lerp(quaternionA.z, quaternionB.z, influence)
+        results.data[i].w = lerp(quaternionA.w, quaternionB.w, factor)
+        results.data[i].x = lerp(quaternionA.x, quaternionB.x, factor)
+        results.data[i].y = lerp(quaternionA.y, quaternionB.y, factor)
+        results.data[i].z = lerp(quaternionA.z, quaternionB.z, factor)
 
     return results
 
-def mixColorLists(VirtualColorList colorsA, VirtualColorList colorsB, VirtualDoubleList influences):
-    cdef long amount = VirtualDoubleList.getMaxRealLength(colorsA, colorsB, influences)
+def mixMatrixLists(VirtualMatrix4x4List matricesA, VirtualMatrix4x4List matricesB, VirtualDoubleList factors):
+    cdef long amount = VirtualDoubleList.getMaxRealLength(matricesA, matricesB, factors)
+    cdef Matrix4x4List results = Matrix4x4List(length = amount)
+    cdef Py_ssize_t i
+
+    for i in range(amount):
+        mixMatrices(results.data + i, matricesA.get(i), matricesB.get(i), <float>factors.get(i))
+
+    return results
+
+def mixColorLists(VirtualColorList colorsA, VirtualColorList colorsB, VirtualDoubleList factors):
+    cdef long amount = VirtualDoubleList.getMaxRealLength(colorsA, colorsB, factors)
     cdef ColorList results = ColorList(length = amount)
     cdef Color *colorA
     cdef Color *colorB
-    cdef float influence
+    cdef float factor
     cdef Py_ssize_t i
 
     for i in range(amount):
-        influence = influences.get(i)
+        factor = <float>factors.get(i)
         colorA = colorsA.get(i)
         colorB = colorsB.get(i)
-        results.data[i].r = lerp(colorA.r, colorB.r, influence)
-        results.data[i].g = lerp(colorA.g, colorB.g, influence)
-        results.data[i].b = lerp(colorA.b, colorB.b, influence)
-        results.data[i].a = lerp(colorA.a, colorB.a, influence)
+        results.data[i].r = lerp(colorA.r, colorB.r, factor)
+        results.data[i].g = lerp(colorA.g, colorB.g, factor)
+        results.data[i].b = lerp(colorA.b, colorB.b, factor)
+        results.data[i].a = lerp(colorA.a, colorB.a, factor)
 
     return results
 
+def mixEulerLists(VirtualEulerList eulersA, VirtualEulerList eulersB, VirtualDoubleList factors):
+    cdef long amount = VirtualDoubleList.getMaxRealLength(eulersA, eulersB, factors)
+    cdef EulerList results = EulerList(length = amount)
+    cdef Euler3 *eulerA
+    cdef Euler3 *eulerB
+    cdef float factor
+    cdef Py_ssize_t i
+
+    results.fill(0)
+    for i in range(amount):
+        factor = <float>factors.get(i)
+        eulerA = eulersA.get(i)
+        eulerB = eulersB.get(i)
+        results.data[i].x = lerp(eulerA.x, eulerB.x, factor)
+        results.data[i].y = lerp(eulerA.y, eulerB.y, factor)
+        results.data[i].z = lerp(eulerA.z, eulerB.z, factor)
+
+    return results
+
+cdef mixMatrices(Matrix4* target, Matrix4* x, Matrix4* y, float factor):
```

Comment:
Can you check this function? when i tried lerping individual elements of matrix directly it produced skewing issues. 
In [blender](https://github.com/blender/blender/blob/743eca0f316d1ec2ef16e6066d7cfc5233c92657/source/blender/blenlib/intern/math_matrix.c#L2428) they decompose matrix into location, quaternion and scaling part. then lerp them individually and compose back to matrix. quaternions are also slerped instead of lerp.

Reply:
Thanks, yes it does when `scale` or `rotation` is mixed.

@OmarEmaraDev We have to split the matrix into components (location, rotation, scale) for mixing. Otherwise, it shows a skewing issue if we `lerp` matrix elements, it should happen because they depend on each other.

Reply:
Ok. Do that.